pkg/cmd/adm/register_member.go

```diff
@@ -120,59 +117,206 @@ func registerMemberCluster(ctx *extendedCommandContext, newCommand client.Comman
+	// the creation logic will be moved to the toolchaincluster_resource controller in toolchain-common and will be based on the secret created above.
+	//
+	// create/update toolchaincluster on the clusterJoinTo
+	term.Printlnf("creating ToolchainCluster representation of %s in %s:", joiningClusterType, clusterJoinToName)
+	toolchainClusterCR := &toolchainv1alpha1.ToolchainCluster{ObjectMeta: metav1.ObjectMeta{Name: toolchainClusterName, Namespace: clusterJoinToOperatorNamespace}}
+	op, err = controllerutil.CreateOrUpdate(context.TODO(), clusterJoinToClient, toolchainClusterCR, func() error {
+
+		// update the tc label
+		labels := toolchainClusterCR.GetLabels()
+		if labels == nil {
+			labels = make(map[string]string)
 		}
-	}()
-	// Create the file
-	file, err := os.CreateTemp("", "add-cluster-*.sh")
+		labels["namespace"] = joiningOperatorNamespace
+		if joiningClusterType == "member" {
+			labels["cluster-role.toolchain.dev.openshift.com/tenant"] = ""
+		}
+
+		toolchainClusterCR.Labels = labels
+		toolchainClusterCR.Spec.APIEndpoint = joiningClusterAPIEndpoint
+		toolchainClusterCR.Spec.SecretRef.Name = secretName
+		if insecureSkipTLSVerify {
+			toolchainClusterCR.Spec.DisabledTLSValidations = []toolchainv1alpha1.TLSValidation{toolchainv1alpha1.TLSAll}
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		return err
+	}
+	term.Println("Toolchaincluster successfully reconciled")
+	term.Printlnf("operation", op)
+	// -- end temporary logic
+
+	return err
+}
+
+func secretName(SANamespacedName runtimeclient.ObjectKey, joiningOperatorNamespace string, joiningClusterName string) string {
+	secretName := SANamespacedName.Name + "-" + joiningOperatorNamespace + "-" + joiningClusterName
+	return secretName
+}
+
+func newRestClient(kubeConfigPath string) (*rest.RESTClient, error) {
+	restClientConfig, err := clientcmd.BuildConfigFromFlags("", kubeConfigPath)
+	restClientConfig.Timeout = 60 * time.Second
+	restClientConfig.ContentConfig = rest.ContentConfig{
+		GroupVersion:         &authv1.SchemeGroupVersion,
+		NegotiatedSerializer: scheme.Codecs,
+	}
+	if err != nil {
+		return nil, err
+	}
+	restClient, err := rest.RESTClientFor(restClientConfig)
 	if err != nil {
 		return nil, err
 	}
-	defer func() {
-		if err := file.Close(); err != nil {
-			term.Printlnf(err.Error())
+	return restClient, nil
+}
+
+// getClusterDetails returns the cluster api endpoint and cluster name
+func getClusterDetails(kubeConfigPath string) (string, string, error) {
+	kubeConfig, err := clientcmd.LoadFromFile(kubeConfigPath)
+	if err != nil {
+		return "", "", err
+	}
+	clusterAPIEndpoint := getServerAPIEndpoint(kubeConfig)
+	clusterURL, err := url.Parse(clusterAPIEndpoint)
+	if err != nil {
+		return "", "", err
+	}
+	// The logic below extracts the domain name from the API server URL, taking everything after "//" until a ":" or "/" (or end of line) is reached.
```

Comment:
This seems to be implementing some of the logic from `utils.GetToolchainClusterName()`.

Reply:
Nice , thanks for the suggestion, I've missed that part. I've now extracted the common logic into a utility function and reused that. 

Take a look at https://github.com/kubesaw/ksctl/pull/53/commits/28e76821c5404538c919fb41cf86e371cf49f45a